app/ui/global-loading.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { useNavigation } from "@remix-run/react";
+import cx from "clsx";
+
+export function GlobalLoading() {
+  const transition = useNavigation();
+  const active = transition.state !== "idle";
+
+  const ref = React.useRef<HTMLDivElement>(null);
+  const [animationComplete, setAnimationComplete] = React.useState(true);
+
+  React.useEffect(() => {
+    if (!ref.current) return;
+    if (active) setAnimationComplete(false);
+
+    Promise.allSettled(
+      ref.current.getAnimations().map(({ finished }) => finished),
+    ).then(() => !active && setAnimationComplete(true));
+  }, [active]);
```

Comment:
I'm all good with these changes, agree it can wait a lil to appear and about the coding style `boolean && callback()` and `import { useEffect } from 'react'` I should have analyzed the repo's style beforehand ;)